### PR TITLE
chore: move Code to a submodule

### LINKED ISSUE #4901: [Chore]: Switch from yarn to a submodule
We currently use yarn link to create a symlink between vendor/modules/code-oss-dev and coder/vscode (our fork). This is a bit unreliable and also requires cloning and updating two separate repos. The first step then would be to switch away from yarn link and use a submodule.

1. Remove vendor
2. Add git submodule under lib/vscode
3. update all the paths and the scripts to point to that location

There may be some additional code to cleanup

### PR DESCRIPTION
**Pull request by code-asher**
Closes #4901.

### CODE REVIEW

**Comment by jsjoeio** [comment]
Looks like CI is failing - [exit code 137](https://github.com/coder/code-server/runs/5544954478?check_suite_focus=true#step:5:4745)

Do we need to [up](https://javahowtos.com/guides/124-docker/418-npm-exit-status-137-when-building-a-docker-image.html) [the RAM](https://github.com/npm/npm/issues/4008#issuecomment-278481967) in CI? 😂

**Comment by code-asher** [comment]
It looks like it is installing dependencies over and over again, I see the same "installing dependencies for..." line multiple times.

**Comment by jsjoeio** [comment]
Docs preview failing again 🤔 

https://github.com/coder/code-server/runs/5544837131?check_suite_focus=true#step:4:18

> Error: Resource not accessible by integration

@BrunoQuaresma did you ever figure out how to fix this? I could have sworn it worked when you and I did it. But Asher's PR is from a fork.

**Comment by BrunoQuaresma** [comment]
Looks like it is a permission issue with the "comment" action, which is weird 
Screen Shot 2022-03-14 at 18 55 01
Looks like it is related to the GITHUB_TOKEN... but I'm not sure if it is needed in this case.

**Review by jsjoeio — COMMENTED** [review]
Me, having `vscode` back in `code-server` repo:

**Comment by jsjoeio** [comment]
I assumed this still runs locally and we've tested a build from CI (the latter I guess can't until CI passes, but we should before merging)

**Review by jsjoeio — APPROVED** [review]

File: ci/dev/fmt.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ main() {
     "*.sh"
   )
   prettier --write --loglevel=warn $(
-    git ls-files "${prettierExts[@]}" | grep -v "lib/vscode" | grep -v "vendor/modules/code-oss-dev" | grep -v 'helm-chart'
+    git ls-files "${prettierExts[@]}" | grep -v "lib/vscode" | grep -v 'helm-chart'
```
[the remark this answers is not in the document]
Comment (code-asher) [reply ci/dev/fmt.sh:22]: lol same

**Comment by code-asher** [comment]
I ran it locally with `yarn watch`, have not ran a build yet though.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -446,12 +445,6 @@ jobs:
           ./test/node_modules/.bin/playwright install-deps
           ./test/node_modules/.bin/playwright install
 
-      # TODO@jsjoeio - remove once we switch to submodules.
-      - name: Create package.json for testing
-        run: |
-          mkdir -p ./vendor/modules/code-oss-dev
-          echo '{ "version": "test" }' > ./vendor/modules/code-oss-dev/package.json
-
```
[the remark this answers is not in the document]
Comment (code-asher) [reply .github/workflows/ci.yaml:454]: Yeah I think the problem was that we cache `node_modules` and then restore that cache.  But Code is under `vendor_modules` so it did not get restored.  Then we tried to read `package.json` from it.

Now that this uses a submodule and we have `submodules: true` in the checkout stage Code will be cloned as part of the checkout phase and `package.json` will exist.

**Comment by code-asher** [comment]
I tested the CI build, all seems well.

**Pull request merged**